reconcilers/reconcilers_test.go

```diff
@@ -605,6 +858,69 @@ func TestChildReconciler(t *testing.T) {
 		ExpectDeletes: []rtesting.DeleteRef{
 			{Group: "", Kind: "ConfigMap", Namespace: testNamespace, Name: testName},
 		},
+	}, {
+		Name: "delete child, clearing finalizer",
+		Parent: resourceReady.
+			MetadataDie(func(d *diemetav1.ObjectMetaDie) {
+				d.Finalizers(testFinalizer)
+			}),
+		GivenObjects: []client.Object{
+			configMapGiven,
+		},
```

Comment:
**[question]**: Can you help me understand why this is a deletion case? I can't tell by looking at the preconditions `Parent` and `GivenObjects`. 🤔

Reply:
this behavior is part of the business logic for the particular reconciler under test. The test case immediate above results in an existing child resource being no longer desired, and deleted without involving a finalizer. This test case is laregly a copy of that test adding in a finalizer.

The delete behavior is driven from https://github.com/vmware-labs/reconciler-runtime/blob/dba890ccfe260716174656751f42836c1c7f8329/reconcilers/reconcilers_test.go#L468-L470